experiment_api/experiment.py

```diff
@@ -0,0 +1,252 @@
+            self.list_of_cameras_per_car[car_id].add(camera_id)
+            self.list_of_cars_per_camera[camera_id].add(car_id)
+            self.list_of_cars[car_id].append(image)
+
+    def __set_target_car(self):
+        # count the number of times distinct cameras spot the car
+        # has to be greater than equal to num_cams, or not drop percentage is going to be higher
+        list_valid_target_cars = []
+        for car_id, camera_ids in self.list_of_cameras_per_car.items():
+            if len(camera_ids) >= self.num_cams:
+                list_valid_target_cars.append(car_id)
+        # return a car id
+        return random.choice(list_valid_target_cars)
+
+    def get_target_car(self):
+        return self.target_car
+
+    def __get_camset(self):
+        num_imgs_per_camset = self.num_cars_per_cam
+        camset = set()
+        # determine whether or not to add the target car
+        if not should_drop(self.drop_percentage):
+            num_imgs_per_camset = num_imgs_per_camset - 1
+            target_car_img = random.choice(self.list_of_cars[self.target_car])
+            camset.add(target_car_img)
+        # grab images
+        random_cam = random.choice(list(self.list_of_cameras_per_car[self.target_car]))
+        for i in range(0, num_imgs_per_camset):
+            random_car = random.choice(list(self.list_of_cars_per_camera[random_cam]))
+            random_car_img = random.choice(self.list_of_cars[random_car])
+            camset.add(random_car_img)
+        # return camset
+        return camset
+
+    def generate(self):
+        self.__set_lists()
+        self.target_car = self.__set_target_car()
+        output = list()
+        for i in range(0, self.num_cams):
+            output.append(self.__get_camset())
+        self.__unset_lists()
+        return output
+
+class Image(object):
+    """ Assume image's name is in the format of carId_cameraId_timestamp_binary.jpg
+    """
+    def __init__(self, img_name, img_dir, img_type):
+        self.name = img_name
+        self.filepath = img_dir + "/" + img_name
+        self.type = img_type
+        self.__splitter = self.name.split("_")
+        self.car_id = self.__set_car_id()
+        self.camera_id = self.__set_camera_id()
+        self.timestamp = self.__set_timestamp()
+        self.binary = self.__set_binary()
+
+    def get_name(self):
+        return self.name
+
+    def __set_car_id(self):
+        car_id = int(self.__splitter[0])
+        return car_id
+
+    def get_car_id(self):
+        return self.car_id
+
+    def __set_camera_id(self):
+        camera_id = int(get_numeric(self.__splitter[1]))
+        return camera_id
+
+    def get_camera_id(self):
+        return self.camera_id
+
+    def __set_timestamp(self):
+        timestamp = datetime.datetime.fromtimestamp(int(self.__splitter[2]))
+        return timestamp
+
+    def get_timestamp(self):
+        return self.timestamp
+
+    def get_timestamp_in_str(self):
```

Comment:
This getter is probably ok, but we could also just calculate that and set it as a value.